File: create_dirs.mjs
#!/usr/bin/env node

import fs from 'node:fs'
import path from 'node:path'

const baseDir = path.resolve(process.cwd());
const dirs = [
    'saas-backend/src/config',
    'saas-backend/src/api/v1/routes',
    'saas-backend/src/api/v1/controllers',
    'saas-backend/src/api/v1/services',
    'saas-backend/src/api/v1/repositories',
    'saas-backend/src/api/v1/validators',
    'saas-backend/src/middleware',
    'saas-backend/src/utils',
    'saas-backend/src/db',
    'saas-backend/sql',
    'saas-backend/tests/unit',
    'saas-backend/tests/integration',
    '.github/workflows',
];

console.log(`Base directory: ${baseDir}`);
console.log('Creating directories...\n');

let successCount = 0;
let errorCount = 0;

dirs.forEach(dir => {
    const fullPath = path.join(baseDir, dir);
    try {
        fs.mkdirSync(fullPath, { recursive: true });
        console.log(`✓ Created: ${dir}`);
        successCount++;
    } catch (error) {
        console.error(`✗ Error creating ${dir}: ${error.message}`);
        errorCount++;
    }
});

console.log(`\n✓ Successfully created ${successCount} directories`);
if (errorCount > 0) {
    console.log(`✗ Failed to create ${errorCount} directories`);
}

File: create_dirs.py
import os
import sys

base_dir = r'c:\Users\gons\OneDrive - BOLT Solutions\Documents\shumagon-goni'
dirs = [
    r'saas-backend\src\config',
    r'saas-backend\src\api\v1\routes',
    r'saas-backend\src\api\v1\controllers',
    r'saas-backend\src\api\v1\services',
    r'saas-backend\src\api\v1\repositories',
    r'saas-backend\src\api\v1\validators',
    r'saas-backend\src\middleware',
    r'saas-backend\src\utils',
    r'saas-backend\src\db',
    r'saas-backend\sql',
    r'saas-backend\tests\unit',
    r'saas-backend\tests\integration',
    r'.github\workflows',
]

os.chdir(base_dir)

for d in dirs:
    try:
        os.makedirs(d, exist_ok=True)
        print(f'✓ Created: {d}')
    except Exception as e:
        print(f'✗ Error creating {d}: {e}')

print('\nAll directories created successfully!')
